### PR TITLE
[T-254] Decorator for checking is_staff on dashboard GET views

### PR DESCRIPTION
**Pull request by lakatop**
Hey,
this isnt a big one, but a tricky one. I checked that decorator is working correctly (e.g. only if user has is_staff flag set it can access view), but I wasnt able to check every single view (as it would probably took me like 2 weeks of testing). So please, @michalpulpan  and @VL-CZ check parts that you are familiar with and ensure that decorator is only on top of **bashboard** GET views. Thank you <3

### CODE REVIEW

File: src/backend/core/category/views.py [hunk 1]
```diff
@@ -90,6 +97,7 @@ class CategoryStorefrontView(APIView):
     Used for storefront.
     """
 
+    @check_user_is_staff_decorator()
```
Comment (VL-CZ) [inline src/backend/core/category/views.py:100]: Remove the decorator from here, cause this view is intended for storefront

File: src/backend/core/category/views.py [hunk 2]
```diff
@@ -110,6 +118,7 @@ class CategoryDetailStorefrontView(APIView):
...
```
Comment (VL-CZ) [inline src/backend/core/category/views.py:121]: same here

File: src/backend/core/category/views.py [hunk 3]
```diff
@@ -135,6 +144,7 @@ class CategoryDetailProductsStorefrontView(APIView):
 
     PRICE_LIST_URL_PARAM = "price_list"
 
+    @check_user_is_staff_decorator()
```
Comment (VL-CZ) [inline src/backend/core/category/views.py:147]: and here 🙂

**Review by VL-CZ — COMMENTED** [review]
Checked `cart`, `category` and `order` views

**Comment by michalpulpan** [comment]
Login on dashboard for `is_staff=false` works, is that correct?

File: src/backend/core/country/views.py [hunk 4]
```diff
@@ -42,6 +44,10 @@ class CountryListView(ListCreateAPIView):
 
     serializer_class = CountrySerializer
 
+    @check_user_is_staff_decorator()
```
Comment (michalpulpan) [inline src/backend/core/country/views.py:47]: Please remove this decorator, it's used on both storefront & dashboard - it's just a GET view.

**Comment by michalpulpan** [comment]
Is it correct that non `is_staff` user can load have data loaded into a view in dashboard? Check currencies for example

File: src/backend/core/category/views.py (hunk 1 above)
Comment (lakatop) [reply src/backend/core/category/views.py]: Fixed.

File: src/backend/core/category/views.py (hunk 2 above)
Comment (lakatop) [reply src/backend/core/category/views.py]: Fixed.

File: src/backend/core/category/views.py (hunk 3 above)
Comment (lakatop) [reply src/backend/core/category/views.py]: Fixed.

File: src/backend/core/country/views.py (hunk 4 above)
Comment (lakatop) [reply src/backend/core/country/views.py]: Fixed.

**Comment by lakatop** [comment]
> Login on dashboard for `is_staff=false` works, is that correct?

How do you mean works? Its not wrapped with decorator, but it shouldnt work for non-staff members as there is check in view.

**Comment by lakatop** [comment]
> Is it correct that non `is_staff` user can load have data loaded into a view in dashboard? Check currencies for example

Not sure what you mean by that. By "loaded", do you mean some `POST`  method?

**Comment by lakatop** [comment]
> Is it correct that non `is_staff` user can load have data loaded into a view in dashboard? Check currencies for example

Not sure what you mean by that. By "loaded", do you mean some `POST`  method?

**Comment by lakatop** [comment]
> Is it correct that non `is_staff` user can load have data loaded into a view in dashboard? Check currencies for example

Not sure what you mean by that. By "loaded", do you mean some `POST`  method?

**Comment by michalpulpan** [comment]
> > Login on dashboard for `is_staff=false` works, is that correct?
> 
> How do you mean works? Its not wrapped with decorator, but it shouldnt work for non-staff members as there is check in view.

I logged into dashboard with user that had `is_staff=false`

**Comment by michalpulpan** [comment]
> Not sure what you mean by that. By "loaded", do you mean some `POST` method?

fetched from backend via `GET` and displayed in table for example

**Comment by lakatop** [comment]
@michalpulpan did you have applied changes from T-254? 
As for login, in view we have `if dashboardLogin is True and not user.is_staff:` check, so I dont have idea how you managed to login with non-staff user 🤔 
For the `GET`, do you know which endpoint was it?

**Comment by michalpulpan** [comment]
> @michalpulpan did you have applied changes from T-254?
> As for login, in view we have `if dashboardLogin is True and not user.is_staff:` check, so I dont have idea how you managed to login with non-staff user 🤔
> For the `GET`, do you know which endpoint was it?

Omg, I'm sorry, for some reason I managed to have `is_staff=true`... Lol
It works like a charm now 😄

**Review by michalpulpan — APPROVED** [review]

**Review by VL-CZ — APPROVED** [review]

**Pull request merged**